compiler/rustc_const_eval/src/const_eval/eval_queries.rs

```diff
@@ -132,8 +133,14 @@ pub(super) fn op_to_const<'tcx>(
         // functionality.)
         _ => false,
     };
-    let immediate = if force_as_immediate && let Ok(imm) = ecx.read_immediate(op) {
-        Right(imm)
+    let immediate = if force_as_immediate {
```

Comment:
I tried it. This would degrade diagnostics significantly, as we wont be able to output an `Option<u32>` as a `Some(99_usize)`, just `Some(Indirect { stuff })`, which IMO is much harder to use.

Reply:
This could be alleviated by equipping the diagnostic code with support for printing integers that are `Indirect`.